app/src/main/kotlin/com/hippo/ehviewer/coil/CropBorderInterceptor.kt

```diff
@@ -0,0 +1,80 @@
+package com.hippo.ehviewer.coil
+
+import android.graphics.Bitmap
+import android.graphics.Rect
+import coil3.BitmapImage
+import coil3.Extras
+import coil3.Image
+import coil3.asCoilImage
+import coil3.getExtra
+import coil3.intercept.Interceptor
+import coil3.intercept.Interceptor.Chain
+import coil3.request.ImageRequest
+import coil3.request.ImageResult
+import coil3.request.SuccessResult
+import coil3.request.allowHardware
+import com.hippo.ehviewer.image.detectBorder
+import com.hippo.ehviewer.util.isAtLeastO
+
+private const val CROP_THRESHOLD = 0.75f
+
+private val maybeCropBorderKey = Extras.Key(default = false)
+
+fun ImageRequest.Builder.maybeCropBorder(enable: Boolean) = apply {
+    extras[maybeCropBorderKey] = enable
+}
+
+val ImageRequest.maybeCropBorder: Boolean
+    get() = getExtra(maybeCropBorderKey)
+
+interface BitmapImageWithRect : Image {
+    val rect: Rect
+    val image: Image
+}
```

Comment:
Why interface instead of class?

Reply:
What's difference here?

Reply:
Can use constructor instead of anonymous object.

Reply:
Why we should not use anonymous object?

Reply:
Cleaner.

Reply:
Never mind, got it

Reply:
Done